src/python_testing/TC_AVSM_2_12.py

```diff
@@ -0,0 +1,294 @@
+            TestStep(20, "TH reads MicrophoneMuted attribute.",
+                     "Verify that the DUT response contains a bool value. Store value as aMicrophoneMuted"),
+            TestStep(21, "TH writes MicrophoneMuted attribute with value set to !aMicrophoneMuted.",
+                     "DUT responds with a SUCCESS status code."),
+            TestStep(22, "TH reads MicrophoneMuted attribute.", "Verify that the value is the same as was written in Step 21."),
+            TestStep(23, "TH reads LocalVideoRecordingEnabled attribute.",
+                     "Verify that the DUT response contains a bool value. Store value as aLocalVideoRecordingEnabled"),
+            TestStep(24, "TH writes LocalVideoRecordingEnabled attribute with value set to !aLocalVideoRecordingEnabled.",
+                     "DUT responds with a SUCCESS status code."),
+            TestStep(25, "TH reads LocalVideoRecordingEnabled attribute.",
+                     "Verify that the value is the same as was written in Step 24."),
+        ]
+
+    @async_test_body
+    async def test_TC_AVSM_2_12(self):
+        endpoint = self.get_endpoint(default=1)
+        cluster = Clusters.CameraAvStreamManagement
+        attr = Clusters.CameraAvStreamManagement.Attributes
+
+        self.step(1)
+        # Commission DUT - already done
+
+        self.step(2)
+        if self.pics_guard(self.check_pics("AVSM.S.A000D")):
+            hdrEnabled = await self.read_single_attribute_check_success(
+                endpoint=endpoint, cluster=cluster, attribute=attr.HDRModeEnabled
+            )
+            logger.info(f"Rx'd HDRModeEnabled: {hdrEnabled}")
+
+        self.step(3)
+        if self.pics_guard(self.check_pics("AVSM.S.A000D")):
+            result = await self.write_single_attribute(attr.HDRModeEnabled(not hdrEnabled),
+                                                       endpoint_id=endpoint)
+            asserts.assert_equal(result, Status.Success, "Error when trying to write HDRModeEnabled")
+            logger.info(f"Tx'd HDRModeEnabled: {not hdrEnabled}")
+
+        self.step(4)
+        if self.pics_guard(self.check_pics("AVSM.S.A000D")):
+            hdrEnabledNew = await self.read_single_attribute_check_success(
+                endpoint=endpoint, cluster=cluster, attribute=attr.HDRModeEnabled
+            )
+            logger.info(f"Rx'd HDRModeEnabled: {hdrEnabledNew}")
+            asserts.assert_equal(hdrEnabledNew, not hdrEnabled, "Value does not match what was written in step 3")
+
+        self.step(5)
+        if self.pics_guard(self.check_pics("AVSM.S.A0013")):
+            softRecordingPrivMode = await self.read_single_attribute_check_success(
+                endpoint=endpoint, cluster=cluster, attribute=attr.SoftRecordingPrivacyModeEnabled
+            )
+            logger.info(f"Rx'd SoftRecordingPrivacyModeEnabled: {softRecordingPrivMode}")
+
+        self.step(6)
+        if self.pics_guard(self.check_pics("AVSM.S.A0013")):
```

Comment:
Again, no need to repeat identical if checks.  Do all the tests for this attribute within a single pics_guard check with an else clause skipping the steps that aren't executed

Reply:
Consolidated under a feature check.